### PR TITLE
Adjust flashlight to more closely match classic scaling

### LINKED ISSUE #20559: osu! flashlight doesn't match stable visually
It probably should. Lazer's implementation looks objectively worse (and provides less visibility).

stable:

Parallels Desktop 2022-10-03 at 07 45 54

lazer:

osu! 2022-10-03 at 07 46 23

### Discussed in https://github.com/ppy/osu/discussions/20553

<div type='discussions-op-text'>

<sup>Originally posted by **Doksha** October  3, 2022</sup>
I like using lazer to practice flashlight as I can change the diameter in game, instead of needing a skin, but since the lazer flashlight is completely different from the stable flashlight, it is hard to get a proper feel for how a section would actually look without going into stable and FCing up to that part; it would be nice if there were an option to emulate the fade and sizes of stable, maybe triggered by the classic mod?</div>

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] https://github.com/ppy/osu-resources/pull/215

Resolves https://github.com/ppy/osu/issues/20559

| classic | lazer |
| --- | --- |
| screenshot001 | osu_2022-10-03_17-57-33 |

It's pretty close. The biggest difference is the brightness in the centre, because it seems like stable's texture is non-linear.

Also changed the combo adjustment - the new values were taken directly from stable.

| combo | classic | lazer |
| --- | --- | --- |
| >100 | screenshot003 | osu_2022-10-03_18-59-48 | 
| >200 | screenshot002 | osu_2022-10-03_18-53-03 |

### CODE REVIEW

**Comment by peppy** [comment]
The brightness difference in the centre feels quite a bit better on stable, and I think we're going to eventually want to match it. Hopefully that won't involve returning to the texture 😅.

That said, doesn't need to be done here (will open a tracking issue).

File: osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs [hunk 1]
```diff
@@ -63,6 +63,7 @@ public OsuFlashlight(OsuModFlashlight modFlashlight)
                 followDelay = modFlashlight.FollowDelay.Value;
 
                 FlashlightSize = new Vector2(0, GetSizeFor(0));
+                FlashlightSmoothness = 1.4f;
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs:66]: osu!catch at very least also seems to use a similar smoothness:

Parallels Desktop 2022-10-03 at 12 23 18

Did you want to look at applying it to other rulesets here or as a separate effort?

**Comment by smoogipoo** [comment]
# Taiko:

(This PR does not fix the offset)

| classic | lazer |
| --- | --- |
| screenshot006 | osu_2022-10-07_14-37-57 |
| screenshot007 | osu_2022-10-07_14-38-21 |
| screenshot008 | osu_2022-10-07_14-38-31 |

# Catch:

| classic | lazer |
| --- | --- |
| screenshot009 | osu_2022-10-07_14-42-27 |
| screenshot010 | osu_2022-10-07_14-42-45 |
| screenshot011 | osu_2022-10-07_14-43-33 |

File: osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs:66]: Done in the following commits.

**Review by peppy — APPROVED** [review]

**Pull request merged**